### ISSUE TITLE
Feature Request - memcopy to already allocated AFArray

### ISSUE THREAD

**Issue by JMurph2015** [JuliaComputing/ArrayFire.jl#158]
Can we get a way to send data to a preallocated location on the GPU?  Does ArrayFire.jl already automatically do this by chance?
Use case:
```julia
using ArrayFire, PortAudio, SampledSignals, DSP
stream = PortAudioStream()
test_data = read(stream, 1s)
output = fft(test_data)
specBuf = AFArray(test_data.data)
@time begin
    for i in 1:1000
        specBuf .= AFArray(test_data.data)
        fft!(specBuf)
        output .= SpectrumBuf(Array(specBuf), nframes(test_data)/samplerate(test_data))
    end
end
```
If in-place "updates" were implemented for AFArrays this algorithm would avoid allocating more GPU memory for every transfer that it sends data over.  Thanks!

**Comment by gaika** [comment]
Definitely possible to copy from GPU to main memory without extra allocation with `af_get_data_ptr`, but don't see how for an opposite direction. @pavanky?

**Comment by gaika** [comment]
possible both ways with `af_write_array` and `af_get_data_ptr`

**Comment by JMurph2015** [comment]
Thanks! That was really useful!  Any gotchas I should be aware of and/or are there docs?
Thanks!  (Also you're doing great work on the Julia 0.6 compatibility!)

**Comment by gaika** [comment]
It is kinda tricky and undocumented on the Julia side, I'll do it. If you want to help, take a look at coverage report, see the functions that are not covered and you think are useful and add a PR with tests here https://github.com/gaika/arrayfire.jl

**Comment by gaika** [comment]
added https://github.com/gaika/ArrayFire.jl/commit/9f8f5ae63a148f0a199fb0a7048dce61cb8df5e6

**Comment by JMurph2015** [comment]
Also @gaika, the thought was in the right place, but as of now the implementation actually ends up with more allocations than the standard way.  Secondly, `x::AFArray .= y::AFArray` currently hits a recursion problem (can't remember right off the top of my head but I want to say it was a stack overflow).  But this is awesome work you're doing. (and I can open this issue on your fork if that's better for you)

**Comment by gaika** [comment]
changed it to use `swap!` instead of copy, should be faster now.

**Comment by gaika** [comment]
merged with [v0.1.0](https://github.com/JuliaComputing/ArrayFire.jl/releases/tag/v0.1.0)
